etna/analysis/forecast/plots.py

```diff
@@ -140,55 +161,63 @@ def plot_forecast(
             forecast_color = line[0].get_color()
 
             # draw prediction intervals from outer layers to inner ones
-            if prediction_intervals and quantiles is not None:
-                alpha = np.linspace(0, 1 / 2, len(quantiles) // 2 + 2)[1:-1]
-                for quantile_idx in range(len(quantiles) // 2):
+            intervals = forecast.get_prediction_intervals()
+            if prediction_intervals and intervals is not None:
+                alpha = np.linspace(0, 1 / 2, len(prediction_intervals_names) // 2 + 2)[1:-1]
+
+                segment_borders_df = intervals.loc[:, pd.IndexSlice[segment, :]].droplevel(level="segment", axis=1)
+                comparator = _get_borders_comparator(segment_borders=segment_borders_df)
+                prediction_intervals_names = sorted(prediction_intervals_names, key=cmp_to_key(comparator))
```

Comment:
Shouldn't we add `ValueError` that can be raised in this line into `Raises` block of `plot_forecast`?

Reply:
Added to docstring